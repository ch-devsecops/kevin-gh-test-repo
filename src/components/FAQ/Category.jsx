import React, { useContext } from 'react';
import {
  Column,
  Copy,
  Box,
  useThemeContext,
  Accordion,
  AccordionExpandAll,
  useAccordion,
} from '@honda-canada/design-system-react';
import keyCodes from '@honda-canada/js-utilities/lib/keyCodes';
import keypressCallback from '@honda-canada/js-utilities/lib/keypressCallback';
import { LayoutContext } from '../LayoutContext';
import { ACURA_THEME_NAME } from '../../utils/constants';

const Category = ({ items, heading, setRef, rendering }) => {
  const header = useThemeContext('header');
  const themeName = useThemeContext('name');
  const { layoutName } = useContext(LayoutContext);
  const isSideNavLayout = layoutName === 'SideNavLayout';
  const leftColumnHeight = isSideNavLayout ? rendering?.leftColumnHeight : 0;
  const mobileAccordionTitleOffset = leftColumnHeight + parseInt(header.mobile.height, 10);
  const behaviour = useAccordion(items);
  const { toggleAllItems, isAllExpanded } = behaviour;

  return (
    <>
      <AccordionExpandAll
        tabIndex={0}
        justifyContent={['flex-end', 'flex-start']}
        alignItems="center"
        flexWrap={['wrap', 'no-wrap']}
        toggleItem={toggleAllItems}
        onKeyDown={keypressCallback(keyCodes.ENTER, toggleAllItems)}
        isActive={isAllExpanded}
        pb={['xs', 'l']}
        setRef={setRef}
        id={heading.key}
        width="100%"
      >
        <Box flexBasis={['100%', 'auto']} mr="auto">
          {heading.title}
        </Box>
        <Copy color={themeName === ACURA_THEME_NAME ? 'primary' : 'black'} fontFamily="bold" lineHeight="16px">
          {isAllExpanded ? heading.collapseText : heading.expandText}
        </Copy>
      </AccordionExpandAll>
      <Column width={[1]} px="zero" pb={['xl', 'big']}>
        <Accordion
          items={items}
          behaviour={behaviour}
          hideBorderOnLast={false}
          isTitleSticky
          titleOffset={[
            mobileAccordionTitleOffset,
            mobileAccordionTitleOffset,
            parseInt(header.desktop.stickyHeight || header.desktop.height, 10),
          ]}
        />
      </Column>
    </>
  );
};

export default Category;
