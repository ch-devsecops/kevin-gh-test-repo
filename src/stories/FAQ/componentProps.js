import mockData from '../../../sample-mock-data/routes/faqs/en.json';

export default mockData;
