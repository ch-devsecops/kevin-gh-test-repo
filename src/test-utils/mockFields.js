/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
