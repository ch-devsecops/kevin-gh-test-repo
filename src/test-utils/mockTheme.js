/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
