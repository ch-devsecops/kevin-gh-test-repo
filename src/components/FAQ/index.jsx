/* eslint-disable react/no-unstable-nested-components */
import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  Column,
  Copy,
  Row,
  MediaItem,
  useOptionalVideo,
  Box,
  Markdown,
  H3,
  Wrapper,
} from '@honda-canada/design-system-react';
import { mapFieldToDesignSystemImage } from '../../utils/sitecoreFields';
import { stripMarkdownHeading, getTitleComponent, styledCompiler } from '../../utils/markdown';
import CTA from '../CTA';
import Category from './Category';

const FAQ = ({ fields, rendering, setRef }) => {
  const { t } = useTranslation();

  if (!fields) {
    return null;
  }

  const { items = [], collapseText, expandText } = fields;

  const headings = items.map(item => {
    const title = item.fields?.categoryTitle.value;
    const TitleComponent = getTitleComponent(title, H3);
    return {
      key: item.fields?.categoryAnchor.value,
      title: <TitleComponent pl={['m', 'zero']}>{styledCompiler(stripMarkdownHeading(title))}</TitleComponent>,
      collapseText: collapseText?.value,
      expandText: expandText?.value,
    };
  });

  const strings = {
    playVideoAria: `${t('Shared.Common.playVideoAria')}`,
  };

  const categories = items.map(category => {
    const questions = category.fields?.items || [];

    return questions.map(question => {
      const SubItemTitle = () => (
        <Copy size="accordionLarge" fontWeight="bold" pl={['m', 'default']}>
          {styledCompiler(stripMarkdownHeading(question.fields?.questionText?.value))}
        </Copy>
      );
      const SubItemContent = () => {
        const video = question.fields?.video || { value: '' };
        const { optionalVideo } = useOptionalVideo({
          src: video.value.href,
          ariaLabel: strings.playVideoAria,
          ...video.value,
        });
        const hasMedia = question.fields?.thumbnailImage?.value.src;
        const image = mapFieldToDesignSystemImage(question.fields.thumbnailImage, true);

        const answerCta = question.fields?.ctaLink?.value?.href ? (
          <Box px={['m', 'default']} pt="default">
            <CTA
              aria-label={t(question.fields?.ctaLink?.value?.text)}
              linkField={question.fields?.ctaLink}
              typeField={question.fields?.ctaType}
              iconField={question.fields?.ctaIcon}
            />
          </Box>
        ) : null;

        if (hasMedia) {
          return (
            <Row>
              <Column width={[1, 1 / 3]} pl={['m', 'default']} pr={['m', 'default']}>
                <Box height="auto" objectFit="cover" mb={['s', 'zero']}>
                  <MediaItem video={optionalVideo} image={image} />
                </Box>
              </Column>
              <Column width={[1, 2 / 3]} pl={['m', 'zero']} pr={['m', 'zero']}>
                <Markdown size="regular">{question.fields.answerText?.value}</Markdown>
              </Column>
              {answerCta}
            </Row>
          );
        }
        return (
          <>
            <Markdown size="regular" px={['m', 'default']}>
              {question.fields.answerText?.value}
            </Markdown>
            {answerCta}
          </>
        );
      };

      return {
        key: question.id,
        id: question.fields.questionAnchor.value,
        title: <SubItemTitle />,
        content: <SubItemContent />,
      };
    });
  });

  return (
    <Wrapper px={0} gutters={[false, false]}>
      <Row>
        {categories.map((category, i) => (
          <Category key={i.toString()} items={category} heading={headings[i]} setRef={setRef} rendering={rendering} />
        ))}
      </Row>
    </Wrapper>
  );
};

export default FAQ;
