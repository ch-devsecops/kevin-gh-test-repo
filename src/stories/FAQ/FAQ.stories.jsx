import React from 'react';
import FAQ from '../../components/FAQ';
import props from './componentProps';

export default {
  title: 'FAQ',
  component: FAQ,
};

const Template = args => <FAQ {...args} />;

export const Default = Template.bind({});
Default.args = props;
