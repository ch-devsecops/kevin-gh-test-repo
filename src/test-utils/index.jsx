import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
