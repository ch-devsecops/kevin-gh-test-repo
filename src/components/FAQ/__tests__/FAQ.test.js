import React from 'react';
import { jssRender, cleanup, axe } from 'test-utils';
import FAQ from '../index';

afterEach(cleanup);

const mockData = require('../__mocks__/mockData.test.json');

describe('<FAQ>', () => {
  test('should render correctly', () => {
    const { container } = jssRender(<FAQ fields={mockData.fields} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should render as null without fields', () => {
    const { container } = jssRender(<FAQ />);

    expect(container.firstChild).toBeNull();
  });

  test('should not initially contain collapse all button', () => {
    const { queryByText } = jssRender(<FAQ fields={mockData.fields} />);
    const { fields } = mockData;

    expect(queryByText(fields.collapseText.value)).not.toBeInTheDocument();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(<FAQ fields={mockData.fields} />);

    expect(await axe(container)).toHaveNoViolations();
  });
});
