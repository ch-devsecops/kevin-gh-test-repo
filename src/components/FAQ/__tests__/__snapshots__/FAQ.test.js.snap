// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAQ> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 kkPryr"
  >
    <div
      class="Box-sc-1bsf784-0 jGVkdm"
      display="flex"
      width="100%"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 eWiJTk sc-1k44dfv-0 bsujjc"
        id="faqCat1"
        tabindex="0"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 ffstyC"
        >
          <h3
            class="sc-1k44dfv-0 cTFHtq"
          >
            <span>
              General FAQs
              <sup>
                tm
              </sup>
            </span>
          </h3>
        </div>
        <p
          class="sc-1k44dfv-0 cApmyO"
          color="black"
          font-family="bold"
          font-weight="regular"
        >
          Expand All
        </p>
        <div
          class="IconWrapper-lp9q2r-0 fIXvxX"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 Column-t2p57o-0 bRfWCx"
        width="1"
      >
        <div>
          <div
            aria-expanded="false"
            aria-pressed="false"
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 lboGYr sc-1k44dfv-0 hPhBSo"
            data-testid="dsr-accordion-item-container"
            id="faqCat1Item1c"
            role="button"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 bPNLGt sc-1k44dfv-0 iCYgbA"
            >
              <p
                class="sc-1k44dfv-0 jQFheS"
                color="typographyDefault"
                font-family="default"
                font-weight="bold"
              >
                <span>
                  Where and how to obtain a radio code?
                  <sup>
                    tm
                  </sup>
                </span>
              </p>
              <div
                class="IconWrapper-lp9q2r-0 fIXvxX"
              />
            </div>
            <div
              aria-live="polite"
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 cVJjVn"
              data-testid="dsr-accordion-item-content-container"
            >
              <div
                class="Box-sc-1bsf784-0 jGVkdm"
                display="flex"
                width="100%"
              >
                <div
                  class="Box-sc-1bsf784-0 Column-t2p57o-0 kxBLEi"
                  width="1,0.3333333333333333"
                >
                  <div
                    class="Box-sc-1bsf784-0 dJluwC"
                    height="auto"
                  >
                    <div
                      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jbBrzI"
                    >
                      <img
                        alt="sample image"
                        class="sc-1k44dfv-0 grcUbM"
                        draggable="true"
                        height="1603"
                        loading="lazy"
                        src="/data/media/img/hero-example.png"
                        width="2400"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="Box-sc-1bsf784-0 Column-t2p57o-0 jQyzvQ"
                  width="1,0.6666666666666666"
                >
                  <div
                    class="sc-1k44dfv-0 klHKZs"
                  >
                    

                    <p>
                      Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. 
                    </p>
                    <ul>
                      <li>
                        You can have lists, like this one

                        <ul>
                          <li>
                            Make things 
                            <strong>
                              bold
                            </strong>
                             or 
                            <em>
                              italic
                            </em>
                          </li>
                          <li>
                            Embed snippets of 
                            <code>
                              code
                            </code>
                          </li>
                          <li>
                            Create 
                            <a
                              href="/"
                            >
                              links
                            </a>
                          </li>
                          <li>
                            ...
                          </li>
                        </ul>
                      </li>
                    </ul>
                    <h6
                      id="ordered-list"
                    >
                      Ordered List
                    </h6>
                    <ol
                      start="1"
                    >
                      <li>
                        You can have lists, like this one
                      </li>
                      <li>
                        Make things 
                        <strong>
                          bold
                        </strong>
                         or 
                        <em>
                          italic
                        </em>
                      </li>
                      <li>
                        Embed snippets of 
                        <code>
                          code
                        </code>
                      </li>
                      <li>
                        Create 
                        <a
                          href="/"
                        >
                          links
                        </a>
                      </li>
                      <li>
                        ...
                      </li>
                    </ol>
                  </div>
                </div>
                <div
                  class="Box-sc-1bsf784-0 dQGyQA"
                >
                  <span>
                    <a
                      class="Link-sc-16wd4a0-0 eYsnbh"
                      href="http://www.acura.ca"
                      rel="noopener noreferrer"
                      target="_blank"
                      title="Some aria label"
                    >
                      My Tertiary CTA
                    </a>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-expanded="false"
            aria-pressed="false"
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 lboGYr sc-1k44dfv-0 hPhBSo"
            data-testid="dsr-accordion-item-container"
            id="faqCat1Item2a"
            role="button"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 bPNLGt sc-1k44dfv-0 iCYgbA"
            >
              <p
                class="sc-1k44dfv-0 jQFheS"
                color="typographyDefault"
                font-family="default"
                font-weight="bold"
              >
                <span>
                  I received a recall notice concerning my vehicle. What should I do now?
                </span>
              </p>
              <div
                class="IconWrapper-lp9q2r-0 fIXvxX"
              />
            </div>
            <div
              aria-live="polite"
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 cVJjVn"
              data-testid="dsr-accordion-item-content-container"
            >
              <div
                class="Box-sc-1bsf784-0 jGVkdm"
                display="flex"
                width="100%"
              >
                <div
                  class="Box-sc-1bsf784-0 Column-t2p57o-0 kxBLEi"
                  width="1,0.3333333333333333"
                >
                  <div
                    class="Box-sc-1bsf784-0 dJluwC"
                    height="auto"
                  >
                    <div
                      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jbBrzI"
                    >
                      <img
                        alt=""
                        class="sc-1k44dfv-0 grcUbM"
                        draggable="true"
                        height="1603"
                        loading="lazy"
                        src="https://app-acura-dit-01-cd.azurewebsites.net/-/media/Acura/modelsCardImages/NSX.jpg?h=1603&w=2400&rev=fa870e20bea14d9e995cda80e2e0ad84&hash=2571A3D47ED59F499F74F879685AF67F"
                        width="2400"
                      />
                      <button
                        aria-label="Shared.Common.playVideoAria"
                        class="IconWrapper-lp9q2r-0 sc-1k44dfv-0 dYhDYF"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="Box-sc-1bsf784-0 Column-t2p57o-0 jQyzvQ"
                  width="1,0.6666666666666666"
                >
                  <div
                    class="sc-1k44dfv-0 klHKZs"
                  >
                    

                    <p>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    </p>
                  </div>
                </div>
                <div
                  class="Box-sc-1bsf784-0 dQGyQA"
                >
                  <a
                    class="Button-pulw6t-0 gIftGV"
                    data-testid="dsr-button-component"
                    href="https://www.honda.ca/"
                    rel="noopener noreferrer"
                    target="_blank"
                    title="Some aria label"
                  >
                    My Secondary CTA
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-expanded="false"
            aria-pressed="false"
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 lboGYr sc-1k44dfv-0 fynLSH"
            data-testid="dsr-accordion-item-container"
            id="faqCat1Item3b"
            role="button"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 bPNLGt sc-1k44dfv-0 iCYgbA"
            >
              <p
                class="sc-1k44dfv-0 jQFheS"
                color="typographyDefault"
                font-family="default"
                font-weight="bold"
              >
                <span>
                  Why should I go to an Acura Dealer for an accident repair?
                </span>
              </p>
              <div
                class="IconWrapper-lp9q2r-0 fIXvxX"
              />
            </div>
            <div
              aria-live="polite"
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 cVJjVn"
              data-testid="dsr-accordion-item-content-container"
            >
              <div
                class="sc-1k44dfv-0 jDhSsD"
                px="m,default"
              >
                

                <p>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
